File: year2/systems-design-implementation/Lab3/Lab3/src/main/java/repo/interfaces/InscriereRepository.java
package repo.interfaces;

import domain.Inscriere;
import domain.Participant;
import domain.Proba;
import utils.Entity;
import utils.Pair;

import java.sql.SQLException;

public interface InscriereRepository<ID, E extends Entity<ID>> extends Repository<Pair<Integer, Integer>, Inscriere> {

    Iterable<Participant> getInscrisi(Proba proba) throws SQLException;

    Integer nrParticipanti(Proba proba) throws SQLException;
}

File: year2/systems-design-implementation/Lab3/Lab3Sharp/Repo/Interfaces/InscriereRepository.cs
using Lab2Sharp.Domain;
using Lab2Sharp.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Repo.Interfaces
{
    interface InscriereRepository<ID, E> : Repository<Pair<int, int>, Inscriere>
    {
        IEnumerable<Participant> GetInscrisi(Proba proba);
        int nrParticipanti(Proba proba);
    }
}
